tests/query_workflow.go

```diff
@@ -175,27 +174,29 @@ func (s *ClientFunctionalSuite) TestQueryWorkflow_QueryWhileBackoff() {
 	s.NotNil(workflowRun)
 	s.True(workflowRun.GetRunID() != "")
 
-	// wait until retry with backoff is scheduled
-	findBackoffWorkflow := false
-	for i := 0; i < 5; i++ {
-		historyEvents := s.getHistory(s.namespace, &commonpb.WorkflowExecution{
-			WorkflowId: id,
-		})
-		if len(historyEvents) == 1 {
-			s.EqualHistoryEvents(`
-  1 WorkflowExecutionStarted {"FirstWorkflowTaskBackoff":{"Seconds":10}}`, historyEvents)
-			findBackoffWorkflow = true
-			break
-		}
+	_, err = s.sdkClient.QueryWorkflow(ctx, id, "", testname)
 
-		// wait for the retry, which will have backoff
-		time.Sleep(time.Second)
+	if expectError {
+		s.Error(err)
+		s.ErrorContains(err, consts.ErrWorkflowTaskNotScheduled.Error())
+	} else {
+		s.NoError(err)
 	}
-	s.True(findBackoffWorkflow)
 
-	_, err = s.sdkClient.QueryWorkflow(ctx, id, "", testname)
-	s.Error(err)
-	s.ErrorContains(err, consts.ErrWorkflowTaskNotScheduled.Error())
+}
+
+func (s *ClientFunctionalSuite) TestQueryWorkflow_QueryWhileBackoff_Pass() {
+	contextTimeout := 10
+	startDelay := 5
+	expectError := false
+	s.queryWorkflow_QueryWhileBackoff(contextTimeout, startDelay, expectError)
```

Comment:
You should combine them in subtests. See `t.Run()` calls in other tests.

Reply:
done in a different way in the next rev